src/test/protocol/KnownFormatToGRPC_test.cpp

```diff
@@ -0,0 +1,924 @@
+                    sfPreviousTxnID.fieldCode,
+                    sfTransactionType.fieldCode,
+                    sfWalletLocator.fieldCode,
+                    sfWalletSize.fieldCode};
+
+                if (excludedTxFields.count(sField.fieldCode) == 1)
+                    continue;
+            }
+
+            // If fmtId is a LedgerEntryType, exclude certain fields.
+            if constexpr (std::is_same_v<KeyType, LedgerEntryType>)
+            {
+                // Fields that gRPC does not include in certain LedgerFormats.
+                //
+                // Some of these may be bugs.
+                static const std::
+                    map<LedgerEntryType, std::vector<SField const*>>
+                        gRPCOmitFields{
+                            {ltACCOUNT_ROOT, {&sfWalletLocator, &sfWalletSize}},
+                            {ltDIR_NODE, {&sfExchangeRate}},
+                            {ltLEDGER_HASHES, {&sfFirstLedgerSequence}},
+                            {ltPAYCHAN, {&sfDestinationNode}},
+                        };
+
+                if (auto const iter = gRPCOmitFields.find(fmtId);
+                    iter != gRPCOmitFields.end())
+                {
+                    std::vector<SField const*> const& omits = iter->second;
+
+                    // Check for fields that gRPC omits from this type.
+                    if (std::find_if(
+                            omits.begin(),
+                            omits.end(),
+                            [&sField](SField const* const omit) {
+                                return *omit == sField;
+                            }) != omits.end())
+                    {
+                        // This is one of the fields that gRPC omits.
+                        continue;
+                    }
+                }
+            }
+
+            // The SFields and gRPC disagree on the names of some fields.
+            // Provide a mapping from SField names to gRPC names for the
+            // known exceptions.
+            //
+            // At least one gRPC spelling is a bug.
+            // clang-format off
+            static const std::map<std::string, std::string> sFieldToGRPC{
+                {"AccountTxnID",      "AccountTransactionID"},
+                {"Fee",               "XRPDropsAmount"},
+                {"PreviousTxnID",     "PreviousTransactionID"},
+                {"PreviousTxnLgrSeq", "PreviousTransactionLedgerSequence"},
+                {"Signature",         "PaymentChannelSignature"},
+                {"SigningPubKey",     "SigningPublicKey"},
+                {"TakerGetsCurrency", "TakerGetsCurreny"},
+                {"TxnSignature",      "TransactionSignature"},
+            };
+            // clang-format on
+
+            std::string gRPCName = sFieldToGRPC.count(sField.getName()) == 1
+                ? sFieldToGRPC.at(sField.getName())
+                : sField.getName();
```

Comment:
This does the lookup twice (once for `count` and once for `at`). We could use `find` instead to save a lookup. I honestly don't care in code like this, it's fine as-is, it just caught my eye.